Type definition in function is not supported

The following program `inline-type.go` triggers an error:

```go
package main

func main() {
        type myint int
        var i = myint(1)
        println(i)
}
```

Expected result:
```console
$ go run ./inline-type.go
1
```

Got:
```console
$ yaegi ./inline-type.go
inline-type.go:5:10: undefined: myint
```

The following program `inline-type-slice.go` triggers a panic:

```go
package main

func main() {
// add a sample
}
```

Expected result:
```console
$ go run ./inline-type-slice.go
1
```

Got:
```console
$ yaegi ./inline-type-slice.go
panic: inline-type-slice.go:5:17: incomplete type

goroutine 1 [running]:
github.com/containous/yaegi/interp.(*itype).TypeOf(0xc0001a4820, 0x0, 0x203000)
        /home/lubr/go/src/github.com/containous/yaegi/interp/type.go:683 +0xb09
github.com/containous/yaegi/interp.(*itype).frameType(0xc0001a4820, 0xd0, 0xc8)
        /home/lubr/go/src/github.com/containous/yaegi/interp/type.go:761 +0x13f
github.com/containous/yaegi/interp.(*itype).frameType(0xc0001a4b60, 0x0, 0x14b9420)
        /home/lubr/go/src/github.com/containous/yaegi/interp/type.go:736 +0x86
github.com/containous/yaegi/interp.(*scope).add(0xc0000b7d00, 0xc0001a4b60, 0xc0000e7500)
        /home/lubr/go/src/github.com/containous/yaegi/interp/scope.go:153 +0x49
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2(0xc0000e7300)
        /home/lubr/go/src/github.com/containous/yaegi/interp/cfg.go:699 +0x8557
github.com/containous/yaegi/interp.(*node).Walk(0xc0000e7300, 0xc00018dd68, 0xc00018dda0)
        /home/lubr/go/src/github.com/containous/yaegi/interp/interp.go:113 +0xaf
github.com/containous/yaegi/interp.(*node).Walk(0xc0000e7100, 0xc00018dd68, 0xc00018dda0)
        /home/lubr/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000e7000, 0xc00018dd68, 0xc00018dda0)
        /home/lubr/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000e6f00, 0xc00018dd68, 0xc00018dda0)
        /home/lubr/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000e6900, 0xc00018dd68, 0xc00018dda0)
        /home/lubr/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000e6400, 0xc00018dd68, 0xc00018dda0)
        /home/lubr/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000e6200, 0xc00018dd68, 0xc00018dda0)
        /home/lubr/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc000088600, 0xc0000e6200, 0xc0000b0658, 0x4, 0x0, 0x0, 0x4)
        /home/lubr/go/src/github.com/containous/yaegi/interp/cfg.go:39 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000088600, 0xc000086070, 0x61, 0x261, 0xc000086070, 0x61, 0x0, 0xc2f260)
        /home/lubr/go/src/github.com/containous/yaegi/interp/interp.go:245 +0x135
main.main()
        /home/lubr/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:77 +0x2e3
```

--- a/_test/fun9.go
+++ b/_test/fun9.go
@@ -1,0 +1,12 @@
+package main
+
+type T uint
+
+func main() {
+	type myint int
+	var i = myint(1)
+	println(i)
+}
+
+// Output:
+// 1

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -309,7 +309,21 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 			return false
 
 		case typeSpec:
-			// processing already done in GTA pass
+			// processing already done in GTA pass for global types, only parses inlined types
+			if sc.def != nil {
+				typeName := n.child[0].ident
+				var typ *itype
+				if typ, err = nodeType(interp, sc, n.child[1]); err != nil {
+					return false
+				}
+				if n.child[1].kind == identExpr {
+					n.typ = &itype{cat: aliasT, val: typ, name: typeName}
+				} else {
+					n.typ = typ
+					n.typ.name = typeName
+				}
+				sc.sym[typeName] = &symbol{kind: typeSym, typ: n.typ}
+			}
 			return false
 
 		case arrayType, basicLit, chanType, funcType, mapType, structType:

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -316,6 +316,10 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 				if typ, err = nodeType(interp, sc, n.child[1]); err != nil {
 					return false
 				}
+				if typ.incomplete {
+					err = n.cfgErrorf("invalid type declaration")
+					return false
+				}
 				if n.child[1].kind == identExpr {
 					n.typ = &itype{cat: aliasT, val: typ, name: typeName}
 				} else {

--- a/interp/gta.go
+++ b/interp/gta.go
@@ -174,7 +174,7 @@ func (interp *Interpreter) gta(root *node, rpath string) ([]*node, error) {
 				n.typ.name = typeName
 				n.typ.path = rpath
 			}
-			// Type may already be declared for a receiver in a method function
+			// Type may be already declared for a receiver in a method function
 			if sc.sym[typeName] == nil {
 				sc.sym[typeName] = &symbol{kind: typeSym}
 			} else {

